discovery-provider/src/__init__.py

```diff
@@ -142,7 +143,7 @@ def create_celery(test_config=None):
     abi_values = helpers.load_abi_values()
     eth_abi_values = helpers.load_eth_abi_values()
     # Initialize eth web
-    eth_web3 = Web3(HTTPProvider(shared_config["web3"]["eth_provider_url"]))
+    eth_web3 = Web3(MultiProvider(shared_config["web3"]["eth_provider_url"]))
```

Comment:
lets explain in comment here why this is applied only to eth_web3 and not web3, someone else coming into this code would instantly be confused